src/gtc/dace/utils.py

```diff
@@ -598,18 +678,42 @@ def __getitem__(self, key: oir.Interval) -> List[Any]:
             raise TypeError("Only OIR intervals supported for keys of IntervalMapping.")
 
         res = []
+        key = oir.UnboundedInterval(start=key.start, end=key.end)
         for start, end, value in zip(self.interval_starts, self.interval_ends, self.values):
-            if key.intersects(oir.Interval(start=start, end=end)):
+            if key.intersects(oir.UnboundedInterval(start=start, end=end)):
                 res.append(value)
         return res
 
 
+def equal_vl_node(n1: "VerticalLoopLibraryNode", n2: "VerticalLoopLibraryNode"):
+    from gtc.dace.nodes import VerticalLoopLibraryNode
+
+    try:
+        assert isinstance(n2, VerticalLoopLibraryNode)
```

Comment:
I think it is difficult to use try/assert to determine this, that way we get different results if `PYTHONOPTIMIZE` is true

Reply:
Thanks, forgot to change this. I pushed it as `if...return False` everywhere now.